src/Common/MemoryBufferWriter.cs

```diff
@@ -217,6 +218,36 @@ public byte[] ToArray()
             return result;
         }
 
+        public int CopyTo(Span<byte> span)
```

Comment:
Why do we need this int other than to do a debug assert? Why can't we use the Length of the stream for that?